skimage/transform/_warps.py

```diff
@@ -106,17 +107,20 @@ def resize(image, output_shape, order=None, mode='reflect', cval=0, clip=True,
                          "dimensions")
 
     if anti_aliasing is None:
-        anti_aliasing = not image.dtype == bool
+        anti_aliasing = (not input_type == bool and
+                         any(x < y for x, y in zip(output_shape, input_shape)))
 
-    if image.dtype == bool and anti_aliasing:
+    if input_type == bool and anti_aliasing:
         warn("Input image dtype is bool. Gaussian convolution is not defined "
              "with bool data type. Please set anti_aliasing to False or "
-             "explicitely cast input image to another data type. Starting "
+             "explicitly cast input image to another data type. Starting "
              "from version 0.19 a ValueError will be raised instead of this "
              "warning.", FutureWarning, stacklevel=2)
 
-    factors = (np.asarray(input_shape, dtype=float) /
-               np.asarray(output_shape, dtype=float))
+    factors = np.divide(input_shape, output_shape)
+    image = convert_to_float(image, preserve_range)
```

Comment:
This conversion to float mustn't be done when `order=0` (please see #5327 :wink:)

Reply:
Thanks for pointing this out. Conflict seems solved !